libpromises/crypto.c

```diff
@@ -502,10 +502,29 @@ size_t CipherTextLengthMax(const EVP_CIPHER* cipher, size_t plaintext_length)
     return plaintext_length + (CipherBlockSizeBytes(cipher) * 2) - 1;
 }
 
+size_t PlainTextLengthMax(const EVP_CIPHER* cipher, size_t ciphertext_length)
+{
+    if(ciphertext_length <= 0)
+    {
+        ProgrammingError("PlainTextLengthMax: ciphertext_length is not positive (%ld)",
+                         ciphertext_length);
+    }
+
+    // check for potential integer overflow, leave some buffer
+    if(ciphertext_length > INT_MAX - 1024)
+    {
+        ProgrammingError("PlainTextLengthMax: ciphertext_length is too large (%ld)",
```

Comment:
If you had other proposals on how to properly "abort" then we can switch if you want.

The thing is that if this function gets called with a buffer that's too small I can't really see how one could recover. The helper functions (e.g. PlaintextLengthMax()) should be used by the caller if uncertain about it will work or not, hence also being a valid programming error IMHO.